### PR TITLE
Always show breadcrumb

### PR DESCRIPTION
**Pull request by richardhj**
The breadcrumb needs to be visible each time, even if there are no records.
This prevents confusion as in https://github.com/terminal42/contao-changelanguage/issues/166#issuecomment-499072164.

//cc @aschempp

### CODE REVIEW

File: core-bundle/src/Resources/contao/themes/flexible/src/main.css [hunk 1]
```diff
@@ -1012,6 +1012,9 @@ label.tl_change_selected {
 	height:16px;
 	vertical-align:-3px;
 }
+#tl_buttons + #tl_breadcrumb {
+	margin:18px 18px 0 18px;
```
Comment (leofeyer) [inline core-bundle/src/Resources/contao/themes/flexible/src/main.css]: Why is this necessary? There should be CSS definitions for `#tl_breadcrumb` already.

Comment (richardhj) [reply core-bundle/src/Resources/contao/themes/flexible/src/main.css]: Because the breadcrumb has no margin. The margin inherits from the tl_formbody_edit, that is not present here.

without the css / with the css definition
Screen Shot 2019-08-21 at 16 44 56
Screen Shot 2019-08-21 at 16 45 11

Comment (richardhj) [reply core-bundle/src/Resources/contao/themes/flexible/src/main.css]: In Contao 4.8 the definition should be `.content > nav` 😅

File: core-bundle/src/Resources/contao/drivers/DC_Table.php [hunk 2]
```diff
@@ -3559,10 +3559,12 @@ protected function treeView()
 			}
 		}
 
+		$breadcrumb = (isset($GLOBALS['TL_DCA'][$table]['list']['sorting']['breadcrumb']) ? $GLOBALS['TL_DCA'][$table]['list']['sorting']['breadcrumb'] : '');
+
 		// Return if there are no records
 		if ($tree == '' && \Input::get('act') != 'paste')
 		{
-			return $return . '
+			return $return.$breadcrumb.'
```
Comment (leofeyer) [inline core-bundle/src/Resources/contao/drivers/DC_Table.php]: Maybe we should wrap the breadcrumb menu with `<div class="tl_formbody_edit"></div>`? Then we would not need extra CSS definitions.

Comment (richardhj) [reply core-bundle/src/Resources/contao/drivers/DC_Table.php]: I was thinking that semantics do not fit when adding that div container.
But I like not changing the css also more.
Changed.

File: core-bundle/src/Resources/contao/drivers/DC_Table.php [hunk 3]
```diff
@@ -3559,10 +3559,12 @@ protected function treeView()
 			}
 		}
 
+		$breadcrumb = (isset($GLOBALS['TL_DCA'][$table]['list']['sorting']['breadcrumb']) ? $GLOBALS['TL_DCA'][$table]['list']['sorting']['breadcrumb'] : '');
+
 		// Return if there are no records
 		if ($tree == '' && \Input::get('act') != 'paste')
 		{
-			return $return . '
+			return $return.'<div class="tl_listing_container">'.$breadcrumb.'</div>
```
Comment (leofeyer) [inline core-bundle/src/Resources/contao/drivers/DC_Table.php]:
```suggestion
			if ($breadcrumb)
			{
				$return .= '<div class="tl_listing_container">' . $breadcrumb . '</div>';
			}

			return $return . '
```

**Review by leofeyer — CHANGES_REQUESTED** [review]
We should only add the markup if the breadcrumb is not empty.

**Comment by leofeyer** [comment]
Note to myself: add the following CSS snippet after merging.

```css
.tl_listing_container + .tl_empty {
    margin-top: -32px;
}
```

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @richardhj.
